### PR TITLE
Support all specialized instances of Vector

### PR DESCRIPTION
**Pull request by RustedBones**
When trying to serialize big scala `Vector`, kryo fails with

```
Cause: com.esotericsoftware.kryo.KryoException: java.lang.IllegalArgumentException: Class is not registered: scala.collection.immutable.Vector3
```

Add support for those types in chill.

Tesing only for `Vector2` as other vectors are too heavy to construct and test

### CODE REVIEW

File: chill-scala/src/main/scala-2.13+/com/twitter/chill/AllScalaRegistrarCompat.scala [hunk 1]
```diff
@@ -32,8 +32,15 @@ private[chill] class AllScalaRegistrarCompat_0_9_5 extends IKryoRegistrar {
...
 private[chill] class AllScalaRegistrarCompat extends IKryoRegistrar {
-  override def apply(newK: Kryo): Unit =
-    newK
-      .forConcreteTraversableClass(Vector[Any]())
-      .forConcreteTraversableClass(Vector('a))
+  override def apply(newK: Kryo): Unit = {
+    // creating actual BigVector is too heavy, just register the class by name
+    val t: TraversableSerializer[Any, Vector[_]] = new TraversableSerializer(true)
+    newK.forConcreteTraversableClass(Vector[Any]())
+    newK.forConcreteTraversableClass(Vector('a)) // Vector1
```
Comment (johnynek) [inline chill-scala/src/main/scala-2.13+/com/twitter/chill/AllScalaRegistrarCompat.scala]: should we make the above follow the below and just do `Vector0` and `Vector1`?

File: chill-scala/src/main/scala/com/twitter/chill/ScalaKryoInstantiator.scala [hunk 2]
```diff
@@ -140,7 +140,7 @@ class ScalaCollectionsRegistrar extends IKryoRegistrar {
       .forTraversableSubclass(ListBuffer.empty[Any], isImmutable = false)
       // add mutable Buffer before Vector, otherwise Vector is used
       .forTraversableSubclass(Buffer.empty[Any], isImmutable = false)
-      // Vector is a final class
+      // Vector is a final class / Vector0
       .forTraversableClass(Vector.empty[Any])
```
Comment (johnynek) [inline chill-scala/src/main/scala/com/twitter/chill/ScalaKryoInstantiator.scala:146]: should we be using `.forTraversableSubclass` here?

Comment (RustedBones) [reply chill-scala/src/main/scala/com/twitter/chill/ScalaKryoInstantiator.scala:146]: I tried but it conflicts with the java wrappers. Change the code comments

**Comment by RustedBones** [comment]
@johnynek I addressed/replied to the comments and did a bit of 'boy-scout' effort, refactoring part of the tests.
Can you have another look ?

File: chill-scala/src/main/scala/com/twitter/chill/ScalaKryoInstantiator.scala [hunk 3]
```diff
@@ -140,7 +140,9 @@ class ScalaCollectionsRegistrar extends IKryoRegistrar {
       .forTraversableSubclass(ListBuffer.empty[Any], isImmutable = false)
       // add mutable Buffer before Vector, otherwise Vector is used
       .forTraversableSubclass(Buffer.empty[Any], isImmutable = false)
-      // Vector is a final class
+      // Vector is a final class in 2.11 / 2.12
+      // for 2.13 we have specific implementations
+      // we should not match subclasses as if conflicts with wrappers
```
Comment (regadas) [inline chill-scala/src/main/scala/com/twitter/chill/ScalaKryoInstantiator.scala]:
```suggestion
      // we should not match subclasses as it conflicts with wrappers
```

**Review by regadas — APPROVED** [review]
lgtm! needs `scalafmt` to have CI green.

**Comment by RustedBones** [comment]
I rebased the PR. That disabled the workflow approval, i was missing some formatting in the scala 2.11/2.12 specialized part (forgot the +)

**Pull request merged**